File: leetcode/matchPlayersAndTrainers.js
var matchPlayersAndTrainers = function(players, trainers) {
    
    players.sort((a, b) => {
       return a - b; 
    });
    
    trainers.sort( (a, b) => {
       return a - b; 
    });
    
    let i = 0, j = 0; 
    let numberOfMatching = 0;
    
    while (i < players.length && j < trainers.length){
        if(players[i] <= trainers[j]){
            numberOfMatching += 1;
            i++;
        }
        j++;
    }
    return numberOfMatching;
};

File: leetcode/matchPlayersAndTrainers.rs
impl Solution {
    pub fn match_players_and_trainers(mut players: Vec<i32>, mut trainers: Vec<i32>) -> i32 {
        
        players.sort();
        
        trainers.sort();
    
        let (mut i, mut j, mut number_of_matching) = (0, 0, 0);
        
        while i < players.len() && j < trainers.len() {
            if (&players[i] <= &trainers[j]){
                i += 1;
                number_of_matching += 1;
            } 
            j += 1; 
        }
        return number_of_matching;
    }
}
